docs/reference/commandline/version.md

```diff
@@ -61,22 +116,71 @@ Server:
   GitCommit:        fec3683
 ```
 
+### API version and version negotiation
+
+The API version used by the client depends on the Docker Engine that the Docker
+CLI is connecting with. When connecting with the Docker Engine, the Docker CLI
+and Docker Engine perform API version negotiation, and select the highest API
+version that is supported by both the Docker CLI and the Docker Engine.
+
+For example, if the CLI is connecting with a Docker 19.03 engine, it downgrades
```

Comment:
This would be interesting to show a table for a few versions ( docker engine v (API v) | docker CLI v (API v) | = API v their will be talking with each other.

Reply:
Oh it exists! Yes, the link would be helpful.